File: robin/python/level2/study2/n_lcm.py
import math


def solution(arr):
    answer = arr[0]

    for i in range(1, len(arr)):
        answer = answer * arr[i] / math.gcd(int(answer), arr[i])

    return answer


if __name__ == '__main__':
    arr = [2, 6, 8, 14]

    print(solution(arr))

File: robin/java/src/programmers/level2/study2/NLcm.java
package programmers.level2.study2;

public class NLcm {

    public static int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }

    public static int lcm(int a, int b) {
        return (a * b) / gcd(a, b);
    }

    public static int solution(int[] arr) {
        int answer = arr[0];

        for (int i = 1; i < arr.length; i++) answer = lcm(answer, arr[i]);

        return answer;
    }

    public static void main(String[] args) {
        int[] arr = {2, 6, 8, 14};
        System.out.println(solution(arr));
    }
}
